File: String/692_top_k_frequent_words.cpp
class Solution {
public:
    vector<string> topKFrequent(vector<string>& words, int k) {
        unordered_map<string, int> map;
        vector<pair<int, string>> h;
        vector<string> result;

        for(string& s: words) {
            map[s] += 1;
        }

        for(auto& p: map) {
            h.push_back({-p.second, p.first});
        }

        make_heap(h.begin(), h.end(), greater<pair<int, string>>());
        for(int i = 0; i < k; i += 1) {
            pop_heap(h.begin(), h.end(), greater<pair<int, string>>());
            result.push_back(h.back().second);
            h.pop_back();
        }

        return result;
    }
};

File: String/692_top_k_frequent_words.py
class Solution:
    def topKFrequent(self, words: List[str], k: int) -> List[str]:
        hashmap = {}
        for word in words:
            hashmap[word] = hashmap.get(word, 0) + 1
        h = [(-count, word) for word, count in hashmap.items()]
        heapq.heapify(h)

        out = []
        while k != 0:
            out.append(heapq.heappop(h)[1])
            k -= 1

        return out